cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,42 @@
+/* eslint-disable max-len */
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
+  let user;
   before(() => {
-
+    cy.task('generateUser').then(generateUser => {
+      user = generateUser;
+    });
+    cy.visit('/automation-practice-form');
   });
 
-  it('', () => {
+  it('should provide an opportunity to register a new user', () => {
+    cy.get('h1')
+      .should('contain.text', 'Practice Form');
+
+    cy.get('#firstName').type(user.firstName);
+    cy.get('#lastName').type(user.lastName);
+    cy.get('#userEmail').type(user.email);
+    cy.contains('.custom-control-label', user.gender).click();
+    cy.get('#userNumber').type(user.mobileNumber);
+    cy.get('#dateOfBirthInput').type('{selectAll}01 Sep 2005{enter}');
+    cy.get('#subjectsContainer').type('Social{enter}');
+    cy.contains('.custom-control-label', user.hobby).click();
+    cy.get('#currentAddress').type('test address, 4');
```

Comment:
```suggestion
    cy.get('#currentAddress').type('test address, 4');
```
you use that variable two times, do not hard code it. Make a variable